File: Módulo Python/Códigos/4 - Estrutura de Repetição/Exercicio.22.py
#22. Faça um programa para escrever os números primos até um número informado pelo usuário. 

limite = int(input("Digite um número inteiro: "))

print(f'Números primos até {limite} são: ')

for num in range(2, limite + 1):
    primo = True
    for i in range(2, num):
        if num % i == 0:
            primo = False
            break
    if primo:
        print(num, end=" ")

File: Livro - Módulo Java/4 - Estrutura de Repetição/Exercicio22.java
import java.util.Scanner;

public class Exercicio22 {
    public static void main(String[] args) {
        // Faça um programa para escrever os números primos até um número informado pelo usuário.

        Scanner scanner = new Scanner(System.in);

        System.out.print("Informe um número limite: ");
        int limite = scanner.nextInt();

        listarPrimos(limite);
    }

    public static boolean ePrimo(int num) {
        if (num <= 1) {
            return false;
        }
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }

    public static void listarPrimos(int limite) {
        if (limite < 2) {
            System.out.println("Por favor, insira um número válido maior ou igual a 2.");
            return;
        }

        System.out.println("Números primos até " + limite + ":");

        for (int i = 2; i <= limite; i++) {
            if (ePrimo(i)) {
                System.out.println(i);
            }
        }
    }
}